packages/extension/src/ui/features/accountNfts/alephium-nft.service.ts

```diff
@@ -103,11 +103,14 @@ async function getNftMetadataz(nftIds: string[], explorerProvider: ExplorerProvi
     }
   }
 
-  const newNftMetadataz = (await Promise.all(
-    chunk(nftIdsWithoutMetadata, maxSizeTokens).map((ids) =>
-      explorerProvider.tokens.postTokensNftMetadata(ids)
-    )
-  )).flat()
+  let newNftMetadataz: NFTMetadata[] = []
+  if (nftIdsWithoutMetadata.length !== 0) {
+    newNftMetadataz = (await Promise.all(
+      chunk(nftIdsWithoutMetadata, maxSizeTokens).map((ids) =>
+        explorerProvider.tokens.postTokensNftMetadata(ids)
+      )
+    )).flat()
```

Comment:
If one of the calls fails, does it mean that none of the successful calls will be cached?

Reply:
Great point, i will update.